UI/src/app/page/use-cases-page/payment-request/containers/payment-page/payment-page.component.spec.ts

```diff
@@ -1,35 +1,38 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { StoreModule } from '@ngrx/store';
-import { PaymentRequestCardComponent } from '../../components/payment-request-card/payment-request-card.component';
-import { SharedTestModule } from './../../../../../shared/components/shared-components.module';
-import { initialState as authInitialState } from './../../../../../store/auth-store/state/init';
-import { initialState as paymentRequestInitialState } from './../../../../../store/payment-request-store/state/init';
-import { initialState as transactionInitialState } from './../../../../../store/transaction-store/state/init';
+import '@angular/compiler';
+import { describe, expect, it, vi } from 'vitest';
 import { PaymentPageComponent } from './payment-page.component';
+import { actionMock, routeMock, snackbarMock, storeMock } from '../../../../../shared/tests/modules/modules.mock';
+import { web3LoginServiceMock } from '../../../../../shared/tests/services/services.mock';
+import { Subject, of } from 'rxjs';
+import { ethereumNetworkMock } from '../../../../../shared/tests/variables/network-detail';
+import { IPaymentRequestState } from 'src/app/store/payment-request-store/state/interfaces';
+import { paymentRequestMock } from '../../../../../shared/tests/variables/payment-request';
 
 describe('PaymentPageComponent', () => {
-  let component: PaymentPageComponent;
-  let fixture: ComponentFixture<PaymentPageComponent>;
+  const paramsObs$ = of({ id: 'mockedId' });
+  routeMock.params = paramsObs$;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      imports: [
-        StoreModule.forRoot({
-          paymentRequest: () => paymentRequestInitialState,
-          transactions: () => transactionInitialState,
-          auth: () => authInitialState
-        }),
-        SharedTestModule
-      ],
-      declarations: [PaymentPageComponent, PaymentRequestCardComponent]
-    }).compileComponents();
-
-    fixture = TestBed.createComponent(PaymentPageComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
+  const component: PaymentPageComponent = new PaymentPageComponent(
+    routeMock,
+    storeMock,
```

Comment:
I can add a test in the case where the user is not login, but this test won't help here because authentication is used in the app-payment-request-card component.